### PR TITLE
Add breaking changes section

### PR DESCRIPTION
**Pull request by fdwr**
It's getting too hard to track the changes and tell others which versions you need to match the current samples and browsers.

FYI @huningxin, @ibelem, @honry

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,28 @@ Please run the following command to download demos required models if you run th
 npm run fetch-models
 ```
 
+### Breaking Changes
+
+WebNN is a living specification and still subject to breaking changes, which may impact the samples depending on your browser version. The following are recent:
+
+- 2024-08-?? ⏳ Add `MLNumber` to `constant`, `clamp`, `pad` - [spec change](https://github.com/webmachinelearning/webnn/pull/647), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5466806)
```
Comment (huningxin) [inline README.md:92]: Would this be a API break change? I mean whether it needs user code change once it is implemented?

File: README.md [hunk 2]
```diff
@@ -85,6 +85,28 @@ Please run the following command to download demos required models if you run th
...
+- 2024-07-22 `argMin`/`argMax` change to take scalar `axis` parameter - [spec change](https://github.com/webmachinelearning/webnn/pull/724]), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5721028)
+- 2024-07-15 `argMin`/`argMax` add `outputDataType` parameter - [spec change](https://github.com/webmachinelearning/webnn/pull/730), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5692538), [sample change](https://github.com/microsoft/webnn-developer-preview/pull/19])
+- 2024-06-12 `softmax` axis argument - [spec change](https://github.com/webmachinelearning/webnn/pull/649), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5495877)
+
+Model                  | Known compatible Chromium version
+-----------------------|-------------------------------
+Segment Anything       | 129.0.6645.0
```
Comment (huningxin) [inline README.md:102]: @ibelem told me 129.0.6617.0 is the minimum Chromium version that works.

File: README.md [hunk 3]
```diff
@@ -85,6 +85,28 @@ Please run the following command to download demos required models if you run th
...
+- 2024-07-24 Allow `MLGraphBuilder.build()` to be called only once - [spec change](https://github.com/webmachinelearning/webnn/pull/717), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5684454), [ORT change](https://github.com/microsoft/onnxruntime/pull/21514), [sample change](https://github.com/microsoft/webnn-developer-preview/pull/21)
+- 2024-07-22 `argMin`/`argMax` change to take scalar `axis` parameter - [spec change](https://github.com/webmachinelearning/webnn/pull/724]), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5721028)
+- 2024-07-15 `argMin`/`argMax` add `outputDataType` parameter - [spec change](https://github.com/webmachinelearning/webnn/pull/730), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5692538), [sample change](https://github.com/microsoft/webnn-developer-preview/pull/19])
+- 2024-06-12 `softmax` axis argument - [spec change](https://github.com/webmachinelearning/webnn/pull/649), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5495877)
+
+Model                  | Known compatible Chromium version
```
Comment (huningxin) [inline README.md:100]: We may add a note that lets people know how to check Chromium version in both Chrome and Edge (Edge as its own version and Chromium version).

**Review by huningxin — COMMENTED** [review]
Thanks @fdwr !

File: README.md (hunk 1 above)
Comment (fdwr) [reply README.md:92]: Ooh, good realization. *Technically* it's an API change, but if BigInt and JS Number implicitly both convert to MLNumber transparently, then the change would be silent for the caller.

File: README.md (hunk 3 above)
Comment (fdwr) [reply README.md:100]: Makes sense given Edge's Chromium version is what actually matter more in this case.

File: README.md (hunk 2 above)
Comment (fdwr) [reply README.md:102]: Is your comment `129.0.6617.0` for just SA or the others too?
(I filled in the Chromium Canary version that I manually verified them all with today)

Comment (ibelem) [reply README.md:102]: @fdwr Based on description in https://github.com/microsoft/webnn-developer-preview/pull/21 , no error for SA and other dev preview demos on Canary 129.0.6617.0 .

File: README.md [hunk 4]
```diff
@@ -85,6 +85,28 @@ Please run the following command to download demos required models if you run th
 npm run fetch-models
 ```
 
+### Breaking Changes
+
+WebNN is a living specification and still subject to breaking changes, which may impact the samples depending on your browser version. The following are recent:
+
+- 2024-08-?? ⏳ Add `MLNumber` to `constant`, `clamp`, `pad` - [spec change](https://github.com/webmachinelearning/webnn/pull/647), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5466806)
+- 2024-07-24 `MLContextOptions::MLPowerPreference` rename `auto` to `default` [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5716629)
+- 2024-07-22 `LSTM`/`GRU` activation enum `MLRecurrentNetworkActivation` - [spec change](https://github.com/webmachinelearning/webnn/pull/718)
```
Comment (mingmingtasd) [inline README.md:94]: [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5689531)

File: README.md [hunk 5]
```diff
@@ -85,6 +85,28 @@ Please run the following command to download demos required models if you run th
...
+- 2024-07-24 Allow `MLGraphBuilder.build()` to be called only once - [spec change](https://github.com/webmachinelearning/webnn/pull/717), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5684454), [ORT change](https://github.com/microsoft/onnxruntime/pull/21514), [sample change](https://github.com/microsoft/webnn-developer-preview/pull/21)
+- 2024-07-22 `argMin`/`argMax` change to take scalar `axis` parameter - [spec change](https://github.com/webmachinelearning/webnn/pull/724]), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5721028)
+- 2024-07-15 `argMin`/`argMax` add `outputDataType` parameter - [spec change](https://github.com/webmachinelearning/webnn/pull/730), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5692538), [sample change](https://github.com/microsoft/webnn-developer-preview/pull/19])
+- 2024-06-12 `softmax` axis argument - [spec change](https://github.com/webmachinelearning/webnn/pull/649), [Chromium change](https://chromium-review.googlesource.com/c/chromium/src/+/5495877)
```
Comment (Honry) [inline README.md:98]: Activation removal is also an API breaking: https://github.com/webmachinelearning/webnn/pull/703/files

**Review by Honry — APPROVED** [review]
👍

**Review by huningxin — APPROVED** [review]
LGTM!

**Pull request merged**